src/main/java/org/spongepowered/api/item/recipe/Recipe.java

```diff
@@ -47,29 +43,18 @@
 public interface Recipe {
 
     /**
-     * Returns the list of item types that result when successful crafting of
-     * this Recipe is completed.
-     *
-     * @return The resultant list of item types
-     */
-    List<ItemType> getResultTypes();
-
-    /**
-     * Checks if the given {@link GridInventory} fits the required constraints
-     * to craft this Recipe.
+     * Gets the result of this {@link Recipe}.
      *
-     * @param grid The ItemGrid to check for validity
-     * @return True if the given input matches this recipe's requirements
+     * @return the results of this {@link Recipe}
      */
-    boolean isValid(GridInventory grid);
+    ImmutableCollection<ItemStackSnapshot> getResults();
 
     /**
-     * Returns the results for running this Recipe over an {@link GridInventory}
+     * Gets the {@link RecipeRegistry} that
+     * corresponds with this Recipe.
      *
-     * @param grid An ItemGrid as input
-     * @return A list of ItemStacks or {@link Optional#empty()} if the given
-     *          ItemGrid does not match this recipe's requirements.
+     * @return the {@link RecipeRegistry} that corresponds with this Recipe
      */
-    Optional<List<ItemStack>> getResults(GridInventory grid);
+    RecipeRegistry<?> getRegistry();
```

Comment:
What is the reason for this method? 

I'm not really against it being here, but IMO its simply not-required overhead.

* If this is not yet/no longer registered, this would return null
* (if ever supported) custom impls might return the wrong registry
* I don't see a usecase

What are your thoughts adding this method?